Sources/Puree/TagPattern.swift

```diff
@@ -19,10 +19,18 @@ public struct TagPattern {
         }
     }
 
-    static func isValidPattern(_ pattern: String) -> Bool {
-        let patternElements = pattern.split(separator: separator)
-        let wildcards = patternElements.filter { $0 == allWildcard || $0 == wildcard }
-        return wildcards.count <= 1
+    private static func isValidPattern(_ pattern: String) -> Bool {
+        enum ValidationError: Error {
+            case invalidElement
+        }
+        
+        let elements = pattern.split(separator: separator, omittingEmptySubsequences: false)
+        
+        let wildcards = try? elements.filter { element in
+            guard !element.trimmingCharacters(in: .whitespacesAndNewlines).isEmpty else { throw ValidationError.invalidElement }
```

Comment:
I think throwing exception doesn't make sense.
You should just return `false` instead of throwing exceptions.

Reply:
+1
and use assertionFailure

Reply:
> I think that's fine too.
> However, interrupt processing asap by throwing error when found an invalid element in my codes.
> Run Sequence.filter andSequence.contains separately is obviously cause the loss of performance.
> I referred to Sequence.first(where:) of Swift stdlib.
> https://github.com/apple/swift/blob/master/stdlib/public/core/Sequence.swift#L991

What do you think about this?